### ISSUE TITLE
The model loading in the demo seems to have become inactive[BUG]

### ISSUE THREAD

**Issue by wan3333** [Harry24k/adversarial-attacks-pytorch#166]
### ✨ Short description of the bug [tl;dr]

The model loading in the demo seems to have become inactive

### 💬 Detailed code and results

`model = load_model('Standard', norm='Linf').to(device)`
It seems cant load the model.

**Comment by rikonaka** [comment]
Hi @wan3333 , if you are a user in mainland China, please add a proxy when running the demo code...because some of the pre-trained models need to be downloaded from the Google driver.

If not, could you please provide the complete code? 😉

**Comment by wan3333** [comment]
Yes, I have proxy,&nbsp;the complete code is in this repository’s demo directory, you can check it.




------------------ Original ------------------
From: isinstance ***@***.***&gt;
Date: Mon,Nov 13,2023 1:35 AM
To: Harry24k/adversarial-attacks-pytorch ***@***.***&gt;
Cc: WAN ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [Harry24k/adversarial-attacks-pytorch] The model loading in thedemo seems to have become inactive[BUG] (Issue #166)

**Comment by rikonaka** [comment]
> Yes, I have proxy,&nbsp;the complete code is in this repository’s demo directory, you can check it.
> […](#)
> ------------------ Original ------------------ From: isinstance ***@***.***&gt; Date: Mon,Nov 13,2023 1:35 AM To: Harry24k/adversarial-attacks-pytorch ***@***.***&gt; Cc: WAN ***@***.***&gt;, Mention ***@***.***&gt; Subject: Re: [Harry24k/adversarial-attacks-pytorch] The model loading in thedemo seems to have become inactive[BUG] (Issue #166)

I can't reproduce the problem you raised.

![ret](https://github.com/Harry24k/adversarial-attacks-pytorch/assets/13602602/e5924f06-27f2-4d20-94ca-7b0e865e0e54)

**Comment by qqt-code** [comment]
I also have this bug

**Comment by rikonaka** [comment]
> I also have this bug

Which specific file are you having trouble with?

**Comment by qqt-code** [comment]
![image](https://github.com/Harry24k/adversarial-attacks-pytorch/assets/63205718/9700bee9-3b08-4314-be97-6cae3743e2fe)
In the demo of White-box Attack on CIFAR10,I download this model，and the content of it is above，so I can't load it.For this problem，i have switched two python version，first one can't download，and second one is the above result. I want to know if it's my operational mistake.

**Comment by rikonaka** [comment]
> ![image](https://private-user-images.githubusercontent.com/63205718/297277550-9700bee9-3b08-4314-be97-6cae3743e2fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0NzQyMTUsIm5iZiI6MTcwNTQ3MzkxNSwicGF0aCI6Ii82MzIwNTcxOC8yOTcyNzc1NTAtOTcwMGJlZTktM2IwOC00MzE0LWJlOTctNmNhZTM3NDNlMmZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDA2NDUxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlMTBmMDZhZDM2YTRkODAwNDdiNjhkMTNjMWFlMTQ1NTA5ZTYzMjU5N2IwOTBmZjVjODQwYjZiMGE5YzNmZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zV2U3RhUdnm8Ih1Yd3E9zTadSIGqXQy2udKT7AR-q8M) In the demo of White-box Attack on CIFAR10,I download this model，and the content of it is above，so I can't load it.For this problem，i have switched two python version，first one can't download，and second one is the above result. I want to know if it's my operational mistake.

May I ask what is the size of your `Standard.pt` file?

I seem to have figured out why, there was a [bug](https://github.com/RobustBench/robustbench/commit/c04877b3f31146c6d4a1c0bd03236a57a27e3f87) in robustbench download function from google driver, and that seems to have just been fixed, but the files included in `torchattacks` were not updated to the latest version of [robustbench](https://github.com/RobustBench/robustbench).

Please delete the `robustbench` folder in `torchattacks` and reclone from this repository https://github.com/RobustBench/robustbench.git, or reinstall it via `pip uninstall robustbench` then `pip install git+https://github.com/RobustBench/robustbench.git` , then it should work 😘.

**Comment by Gorgerbin** [comment]
> ![image](https://private-user-images.githubusercontent.com/63205718/297277550-9700bee9-3b08-4314-be97-6cae3743e2fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MjIxMDksIm5iZiI6MTcwNzgyMTgwOSwicGF0aCI6Ii82MzIwNTcxOC8yOTcyNzc1NTAtOTcwMGJlZTktM2IwOC00MzE0LWJlOTctNmNhZTM3NDNlMmZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDEwNTY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhMjY1MjU0NzcyMzA4NWE0ZWRkMTFiZmNkYjQ1YmE1ZjczMzMzNmNiODE3NDcwNTIwMWM4YTgwYjJjMTY5MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ru1rNeulmIXFIwuKyxQJ594U0Bhdy1MiB9VGWYdlDbE) In the demo of White-box Attack on CIFAR10,I download this model，and the content of it is above，so I can't load it.For this problem，i have switched two python version，first one can't download，and second one is the above result. I want to know if it's my operational mistake.

do you resolve this problem please?

**Comment by rikonaka** [comment]
Hi @Gorgerbin @qqt-code @wan3333 , it seems that the problem is caused by the `robustbench` library, so until the problem is fixed upstream, you can test it with this new demo code:
https://colab.research.google.com/drive/1atyo82sYeaZsbudgjt7jo25Cuv6ArzKB?usp=sharing
or here: 
[new_demo.zip](https://github.com/Harry24k/adversarial-attacks-pytorch/files/14274887/new_demo.zip)

This requires three files, one for the trained model (`resnet18_eval.pth`) and two for the image and label (`images.pth` and `labels.pth`). These three files you can download from here:

* model
https://github.com/rikonaka/adversarial-attacks-pytorch/raw/master/code_coverage/resnet18_eval.pth

* images
https://github.com/rikonaka/adversarial-attacks-pytorch/raw/master/code_coverage/images.pth

* labels
https://github.com/rikonaka/adversarial-attacks-pytorch/raw/master/code_coverage/labels.pth

😘😘😘

**Comment by Gorgerbin** [comment]
briefly thank you for taking the time to address my question！


Zebin&nbsp;Guo
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Harry24k/adversarial-attacks-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年2月13日(星期二) 晚上11:53
***@***.***&gt;;
抄送:&nbsp;"Zebin ***@***.******@***.***&gt;;
主题:&nbsp;Re: [Harry24k/adversarial-attacks-pytorch] The model loading in the demo seems to have become inactive[BUG] (Issue #166)





 
Hi @Gorgerbin , it seems that the problem is caused by the robustbench library, so until the problem is fixed upstream, you can test it with this demo code.
 
https://colab.research.google.com/drive/1s_fclW-pMgKW6UzPV8agV2lBq96rPQsS?usp=sharing
 
This requires three files, one for the trained model (resnet18_eval.pth) and two for the image and label (images.pth and labels.pth). These three files you can download from here (https://github.com/rikonaka/adversarial-attacks-pytorch/tree/master/code_coverage).

**Comment by pingguokiller** [comment]
I have the same problem.

In 'robustbench/utils.py': 'response = session.get(url_base, params={'id': gdrive_id}, stream=True)',
when downloading the .pt  file, google would give the warning info to ask you to confirm. So it downloads a file with html instead of the real pt file.

![1712756824936](https://github.com/Harry24k/adversarial-attacks-pytorch/assets/10582481/048a5f1b-704b-44eb-8588-c3631dfc83db)

**Comment by pingguokiller** [comment]
> I have the same problem.
> 
> In 'robustbench/utils.py': 'response = session.get(url_base, params={'id': gdrive_id}, stream=True)', when downloading the .pt file, google would give the warning info to ask you to confirm. So it downloads a file with html instead of the real pt file.
> 
> ![1712756824936](https://private-user-images.githubusercontent.com/10582481/321261563-048a5f1b-704b-44eb-8588-c3631dfc83db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3NTczODcsIm5iZiI6MTcxMjc1NzA4NywicGF0aCI6Ii8xMDU4MjQ4MS8zMjEyNjE1NjMtMDQ4YTVmMWItNzA0Yi00NGViLTg1ODgtYzM2MzFkZmM4M2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEwVDEzNTEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5ZmM1MTE0MTkzNWNlNjljNmQ2NzIwNzk3YzAxMzYwNWYwZjRiYzkyZjQ1NzRiZGY0MTFmN2I3YTE3Mzg1ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.K5VUqpcCZ698vcCXtWJ0vA3z2T83Ur0Pv4rZDjNnWmE)

You can manually download the .pt file with url "https://docs.google.com/uc?export=download&confirm=t&id={gdrive_id}".
